cli/flox-rust-sdk/src/models/manifest/typed.rs

```diff
@@ -806,6 +810,34 @@ pub enum ManifestError {
     PkgOrGroupNotFound(String),
 }
 
+/// The section where users can declare dependencies on other environments.
+#[derive(Debug, Clone, Serialize, Deserialize, Default, PartialEq, Eq)]
+#[cfg_attr(test, derive(proptest_derive::Arbitrary))]
+pub struct Include {
+    pub environments: Vec<IncludeDescriptor>,
+}
+
+impl Include {
+    pub(crate) fn skip_serializing(&self) -> bool {
+        self.environments.is_empty()
+    }
+}
+
+/// The structure for how a user is able to declare a dependency on an environment.
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+#[cfg_attr(test, derive(proptest_derive::Arbitrary))]
+#[serde(untagged)]
+pub enum IncludeDescriptor {
+    Local {
+        /// The directory where the environment is located.
+        dir: PathBuf,
+        /// A name similar to an install ID that a user could use to specify
+        /// the environment on the command line e.g. for upgrades, or in an
+        /// error message.
+        name: String,
```

Comment:
(non-blocking, style) `name` before `dir` (in the struct and examples) feels more natural to me. I assume we can't use a HashMap because they need to be ordered.

Reply:
Right, we can’t use a map because the environments need to be ordered.

Reply:
TOML maps are not ordered, so this would be a _convention_ we're adding on top of TOML syntax. I think that's probably as bad as implementing logic in TOML, so let's not do that even if it's possible.

Reply:
Do we need to validate that names are unique?

I kinda forgot that we have a name in `env.json`. Would it be better to pull out the name for every included environment, and then error and require `name` only if there's a collision? We could also do that for remote environments, if e.g. I included both `owner1/name` and `owner2/name` then we could error.

I also think this is okay for now

Reply:
I'm not sure what you mean by "pull out the name". I still think you have the issue of the environment name not being terribly user visible. I don't mind making this field an optional override for the collisions like you mentioned (as a follow up), but since we were planning to embed the include descriptor in the lockfile I'd like to nail that down soon.

Reply:
e.g. do we include an optional `name` field for both local and remote include descriptors. When we actually _use_ the `name` field is up to us to decide, but I'd like to have the lockfile format settled since changing the lockfile format causes a lot of uncertainty once we've started pushing environments to FloxHub and stuff like that (i.e. I'd like to avoid "pull -> error -> oh this must have been an old environment of mine from when we were working on composition").

Reply:
By pull out the name, I meant for path environments get the name out of env.json, and for remote environments use the `name` component of `owner/name`. I think it could work to use those unless there were conflicts. And name could be optional for both remote and dir includes, and the way to resolve conflicts